### PR TITLE
added MatrixGENReader

### PR DESCRIPTION
**Pull request by cseed**

### CODE REVIEW

File: hail/src/main/scala/is/hail/io/gen/LoadGen.scala [hunk 1]
```diff
@@ -95,3 +98,84 @@ object LoadGen {
...
+  files: Array[String],
+  sampleFile: String,
+  chromosome: Option[String],
+  nPartitions: Option[Int],
+  tolerance: Double,
+  rg: Option[String],
+  contigRecoding: Map[String, String],
+  skipInvalidLoci: Boolean) extends MatrixReader {
+
+  println(files.toFastIndexedSeq)
```
Comment (jbloom22) [inline hail/src/main/scala/is/hail/io/gen/LoadGen.scala]: remove

File: hail/src/main/scala/is/hail/io/gen/LoadGen.scala [hunk 2]
```diff
@@ -95,3 +98,84 @@ object LoadGen {
...
+  private val unequalSamples = results.filter(_.nSamples != nSamples).map(x => (x.file, x.nSamples))
+  if (unequalSamples.nonEmpty)
+    fatal(
+      s"""The following GEN files did not contain the expected number of samples $nSamples:
+         |  ${ unequalSamples.map(x => s"""(${ x._2 } ${ x._1 }""").mkString("\n  ") }""".stripMargin)
+
+  private val noVariants = results.filter(_.nVariants == 0).map(_.file)
+  if (noVariants.nonEmpty)
+    fatal(
+      s"""The following GEN files did not contain at least 1 variant:
```
Comment (jbloom22) [inline hail/src/main/scala/is/hail/io/gen/LoadGen.scala]: "The following GEN files contained no variants"

File: hail/src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala [hunk 3]
```diff
@@ -297,16 +297,6 @@ class MatrixIRSuite extends SparkSuite {
...
-    val gen = hc.importGen("src/test/resources/example.gen", "src/test/resources/example.sample",
-      contigRecoding = Some(Map("01" -> "1")))
-    val genIR = MatrixMapCols(gen.ast, SelectFields(InsertFields(Ref("sa", gen.colType),
-      FastIndexedSeq("id1" -> Str(""), "id2" -> Str(""), "missing" -> F64(0.0))), FastIndexedSeq("id1", "id2", "missing")),
-      Some(FastIndexedSeq("id1")))
-    val path = tmpDir.createLocalTempFile()
-    Interpret(MatrixWrite(genIR, MatrixGENWriter(path)))
-  }
-
   @Test def testMatrixMultiWrite() {
```
Comment (jbloom22) [inline hail/src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala]: is the new path being tested?

Comment (cseed) [reply hail/src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala]: Yes, see test_impex.py:GENTests.

**Pull request merged**